src/redux/selectedThemeSlice.ts

```diff
@@ -0,0 +1,38 @@
+import { Theme, createTheme } from '@mui/material';
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { themes } from '../utils/themes';
+
+const OSTheme = window
+  .matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+
+localStorage.setItem('OSTheme', OSTheme);
+
+const getTheme = (themeName: string): Theme => {
+  const theme = themes.find((foundTheme) => foundTheme.themeName === themeName)
+    || { themeName: '', themeObject: createTheme() };
+
+  return createTheme(theme.themeObject);
+};
+
+const themeNameFromLocalStorage = localStorage.getItem('theme');
+
+const initialState: Theme = themeNameFromLocalStorage
+  ? getTheme(themeNameFromLocalStorage)
+  : getTheme(OSTheme);
+
+const selectedThemeSlice = createSlice({
+  name: 'selectedTheme',
+  initialState,
+  reducers: {
+    selectTheme: (_theme, action: PayloadAction<string>) => {
+      const newTheme = getTheme(action.payload);
+
+      localStorage.setItem('theme', action.payload);
+
+      return newTheme;
```

Comment:
You are not updating your state in your reducer. It's useless to return some data from the reducer.

Reply:
state updating when you returning some value from redux, this is a feature of redux toolkit.

Reply:
Ok. But consider to do the next things: 
1) remove side effects from your reducer (local storage writing)
2) It's a more undestandable and common to update state in redux toolkit like this way:
(state, action) => state.targetKey = action.payload
That's convinient and allows you to avoid reseiving unused state, as a first parameter, and give it the name like _name